src/components/FilterSection.tsx

```diff
@@ -1,25 +1,97 @@
 import React from "react";
+import { useDispatch, useSelector } from "react-redux";
 import Grid from "@mui/system/Grid";
-import { DateFilter, GenderFilter, AgeRangeFilter } from "./shared";
+import {
+  CheckboxFilter as GenderFilter,
+  MultiSelectFilter as NameMultiSelect,
+  DateFilter as DateOfBirthFilter,
+  RangeFilter as AgeRangeFilter,
+  Button,
+} from "./shared";
+import {
+  setFilterUsername,
+  setGenderFilter,
+  setFilterDateOfBirth,
+  setAgeRangeFilters,
+  reset,
+} from "../slices";
 
 const FilterSection: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const genderOptions = [
+    { label: "Male", value: "male" },
+    { label: "Female", value: "female" },
+  ];
+
+  const nameFilterOptions = useSelector((state) => state.users.nameFilters);
+
+  const dateFormat = "YYYY-M-D";
+
+  const markers = {
+    18: "18",
+    25: "25",
+    35: "35",
+    45: "45",
+    55: "55",
+    65: "65",
+  };
+  const ageRangeOptions = { markers: markers, min: 18, max: 65 };
+
+  const buttonOptions = {
+    text: "Reset",
+    variant: "outlined",
+    colour: "danger",
+    icon: "ClearOutlined",
+  };
+
+  const handleOnChange = (type: string, value: unknown) => {
```

Comment:
instead of keeping individual slice for each filter we can maintain on slice to keep filter data. 
filters: { selectedGender: 'M', selectedUsername: '', ...  etc  }

Reply:
Not sure on how to do this